src/lib/proxy/proxy-repository.ts

```diff
@@ -88,11 +88,15 @@ export class ProxyRepository
         clearTimeout(this.timer);
     }
 
-    private async loadDataForToken() {
+    private async dataPolling() {
         this.timer = setTimeout(async () => {
-            await this.loadDataForToken();
+            await this.dataPolling();
         }, this.randomizeDelay(this.interval, this.interval * 2)).unref();
 
+        await this.loadDataForToken();
```

Comment:
why does this solve the problem?

Reply:
recursive??

Reply:
We made some separation of concerns, since `onAnyEvent` only calls `loadDataForToken`, not `dataPolling`, we no longer create new timers when events happen: https://github.com/Unleash/unleash/blob/96e1758b872ac7e4d23cf6c0b29668dad3efc73e/src/lib/proxy/proxy-repository.ts#L112-L114